app/controllers/shopify_app/sessions_controller.rb

```diff
@@ -36,10 +36,14 @@ def destroy
 
     private
 
-    def authenticate
+    def authenticate      
       if sanitized_shop_name.present?
         session['shopify.omniauth_params'] = { shop: sanitized_shop_name }
-        fullpage_redirect_to "#{main_app.root_path}auth/shopify"
+        if session['shopify.cookies_persist'] || !ShopifyApp.configuration.embedded_app?
```

Comment:
I think there's a subtle bug here.

When we're in fullpage mode, we might already have the `shopify.cookies_persist` cookie, which would mean we'd take this path. But the whole reason we're in fullpage mode is that we want to trigger a "fullpage redirect", since that's where our ITP logic currently lives.

We should separate these two concerns by fullpage redirecting to a `set_top_level_cookie` action or passing a similarly-named parameter that will always trigger the top-level cookie (ITP) logic.